node/internal/child_process.ts

```diff
@@ -411,6 +446,173 @@ function normalizeStdioOption(
   }
 }
 
+export function normalizeSpawnArguments(
+  file: string,
+  args: string[],
+  options: SpawnSyncOptions,
+) {
+  validateString(file, "file");
+
+  if (file.length === 0) {
+    throw new ERR_INVALID_ARG_VALUE("file", file, "cannot be empty");
+  }
+
+  if (ArrayIsArray(args)) {
+    args = ArrayPrototypeSlice(args);
+  } else if (args == null) {
+    args = [];
+  } else if (typeof args !== "object") {
+    throw new ERR_INVALID_ARG_TYPE("args", "object", args);
+  } else {
+    options = args;
+    args = [];
+  }
+
+  if (options === undefined) {
+    options = kEmptyObject;
+  } else {
+    validateObject(options, "options");
+  }
+
+  let cwd = options.cwd;
+
+  // Validate the cwd, if present.
+  if (cwd != null) {
+    cwd = getValidatedPath(cwd, "options.cwd") as string;
+  }
+
+  // Validate detached, if present.
+  if (options.detached != null) {
+    validateBoolean(options.detached, "options.detached");
```

Comment:
For the options that are not currently supported, I wonder if it makes sense to perform the validation so that the input is correct, but then immediately throw an exception saying that the option is unimplemented so that users don't get unexpected behavior.

Reply:
Agreed. I’ll do this in another PR (likely the one that fixes typings around these functions, as previously mentioned).